src/core/jpg.js

```diff
@@ -1019,7 +1019,16 @@ var JpegImage = (function jpegImage() {
       // type of data: Uint8Array(width * height * numComponents)
       var data = this._getLinearizedBlockData(width, height);
 
-      if (this.numComponents === 3) {
+      if (this.numComponents === 1 && forceRGBoutput) {
+        var rgbData = new Uint8Array(width * height * 3);
+        var offset = 0;
+        for (var i = 0, ii = data.length; i < ii; i++) {
+          rgbData[offset++] = data[i];
```

Comment:
I'm not sure if the jit is smart enough to realize it's accessing the same data[i] here. Best to probably stick it in a variable within the loop.